opensrp-app/src/main/java/org/smartregister/sync/helper/LocationServiceHelper.java

```diff
@@ -78,6 +79,19 @@ public static LocationServiceHelper getInstance() {
     }
 
     protected List<Location> syncLocationsStructures(boolean isJurisdiction) {
+        List<Location> locationStructures = batchSyncLocationsStructures(isJurisdiction);
+        int batchFetchCount = locationStructures.size();
+
+        while( locationStructures != null &&  batchFetchCount >= LOCATION_PULL_LIMIT) {
```

Comment:
since the limit can always be changed from the server, retry sync if any record was synced